### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/opencbdc-tx/commit/0659f899e5143d14b1036b546bd6b84e853cb85b#commitcomment-66429463

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/opencbdc-tx/commit/a223cebba0576c4ed6c594aba2bd010a410e3bd3

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

I intend to annotate this PR shortly. I'm happy to drop changes to files, drop specific terms, or split things as requested, although this may take some time.

### CODE REVIEW

File: src/uhs/atomizer/shard/shard.hpp [hunk 1]
```diff
@@ -61,7 +61,7 @@ namespace cbdc::shard {
         /// \return true if the shard successfully digested the block. False if the block height is not contiguous.
         auto digest_block(const cbdc::atomizer::block& blk) -> bool;
 
-        /// Returns the height of the the most recently digested block.
+        /// Returns the height of the most recently digested block.
```
Comment (jsoref) [inline src/uhs/atomizer/shard/shard.hpp:64]: This isn't strictly a spelling fix, but old fashioned spell checkers had support for this flag, so I've been prototyping it.

File: src/uhs/client/client-cli.cpp [hunk 2]
```diff
@@ -281,7 +281,7 @@ auto main(int argc, char** argv) -> int {
             return -1;
         }
     } else {
-        std::cerr << "Unkown command" << std::endl;
+        std::cerr << "Unknown command" << std::endl;
```
Comment (jsoref) [inline src/uhs/client/client-cli.cpp:284]: Sometimes output changes are API breaks.

**Comment by HalosGhost** [comment]
I think we would be happy to merge this (better readability is definitely an improvement). However, merging each commit does not make sense to me. This is an omnibus PR to fix some spelling mistakes. It should be probably be squashed to a single commit with a descriptive message (e.g., `docs: fix various typos`).

**Comment by HalosGhost** [comment]
@jsoref please run `clang-format` on the code per the lint CI check, squash and force-push.

File: src/uhs/client/client-cli.cpp (hunk 2 above)
Comment (HalosGhost) [reply src/uhs/client/client-cli.cpp:284]: We don't make any guarantee of a stable API (this is a research project; and that will necessitate changes), so I'm not too worried about this.

File: src/uhs/atomizer/shard/shard.hpp (hunk 1 above)
Comment (HalosGhost) [reply src/uhs/atomizer/shard/shard.hpp:64]: Still a typo and a good opportunity to improve readability. Thank you for flagging it.

**Review by HalosGhost — APPROVED** [review]
This fixes a lot of typos (both in documentation/comments, and in the code itself). Simple and easy.

Merging.

**Pull request merged**